src/tests/complex_tests/mod.rs

```diff
@@ -331,24 +366,34 @@ fn run_and_try_create_witness_inner(
     LocalFileDataSource::create_folders_for_storing_data();
     use crate::data_source::*;
 
+    let circuits_len = basic_block_circuits.len();
+
     for (idx, el) in basic_block_circuits.clone().into_iter().enumerate() {
         let descr = el.short_description();
-        println!("Doing {}: {}", idx, descr);
+        println!("Doing {} / {}: {}", idx, circuits_len, descr);
 
         if el.numeric_circuit_type() != previous_circuit_type {
             instance_idx = 0;
         }
 
-        if let Ok(proof) = source.get_base_layer_proof(el.numeric_circuit_type(), instance_idx) {
-            if instance_idx == 0 {
-                source.set_base_layer_padding_proof(proof).unwrap();
-            }
+        if options.reuse_artifacts {
```

Comment:
You can't really reuse artifcats if not available.

```suggestion
        if options.try_reuse_artifacts {
```